### DISCUSSION TITLE
Migrating away from symfony1: strategies, tricks, ideas

### DISCUSSION THREAD

**Discussion by mentalstring** [FriendsOfSymfony1/symfony1#324] [General]
I think at least a fair share of those still using symfony1 would like to replace it with something more modern but, despite it's name, Symfony2+ is a very different framework, so migration away from symfony1 is quite a big task that is often hard to justify for large projects.

Nonetheless, I think this is a topic that interests some, so I'm starting a discussion on it so that there's a place to share experiences and ideas on that.

Over the years I collected a few links related to this, mostly in the form of symfony2+ wrappers around a symfony1 project:

* https://blog.theodo.com/2015/01/wrap-up-your-legacy-application-with-symfony/
* https://github.com/theodo/TheodoEvolutionLegacyWrapperBundle
* https://github.com/theodo/TheodoEvolutionSessionBundle
* https://github.com/Ingewikkeld/IngewikkeldWrapperBundle
* https://github.com/LeaseWeb/LswRemoteTemplateBundle
* https://github.com/marfillaster/ButterweedSF1EmbedderBundle
* https://blog.fingo.pl/migration-symfony-1-4-to-symfony-4-4-using-strangler-pattern/
* https://github.com/dkusmierek/symfony_migration_example
* http://www.slideshare.net/futurecat/nice-performance-using-sf2-cache-wrapping-sf1-application-paris

and some to wrap any generic legacy project:

* https://symfony.com/doc/current/migration#front-controller-with-legacy-bridge
* https://www.youtube.com/watch?v=YzyiZNY9htQ
* https://github.com/gertvdb/LegacyBridgeBundle

Now most links all quite outdated, but perhaps they can still be useful for ideas and strategies on how go about it.

With this, have you tried (successfully or not) to migrate a project built with symfony1? What were the main hurdles you found? How long did it take?

**Comment by connorhu** [comment]
I am working on two legacy codes at the moment, neither of them symfony1, but both will be symfony7. The first one is a 20 year old code, organized in files but with a shared scope. Smarty, Adodb and some service classes (Class named constructors!!! hoooly moly).
The second one is organized in classes, but it is quite hard to see the logic.

In both cases I define services and push the application logic towards ServiceSubscriber, with Request input and Response output. Once that's done, symfony will eat it without any problem whatever the layout.

**Comment by mcgrogan91** [comment]
Well this is certainly timely.  The application I work on is on 1.4 and we've talked for years about eventually upgrading, but there wasn't a good path for it.  We decided to carve some time out in March/April to really devote to figuring out what our path forward is, and this is a great set of resources.  I'm excited to see some existing Bundles to try and mess with!
